src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2945,14 +2945,28 @@ private String getSignatureForInstanceTTEFn(
       String classTemplatizedType =
           className.equals("ಠ_ಠ.clutz.goog.Promise") ? " any" : className + " < RESULT >";
       if (propName.equals("then")) {
-        return "then < RESULT > (opt_onFulfilled ? : ( (a : "
-            + templateVarName
-            + " ) => "
-            + classTemplatizedType
-            + " | RESULT ) | null , "
-            + "opt_onRejected ? : ( (a : any ) => any ) | null) : "
-            + classTemplatizedType
-            + " ;";
+        // Prune promise<never> off AngularJS IPromise.then return type.
```

Comment:
This comment is too specific. Describe the general aim here - match typings of angular.js.

Reply:
Done.